src/Airship/Resource.hs

```diff
@@ -27,7 +27,7 @@ data PostResponse s m
     = PostCreate [Text] -- ^ Treat this request as a PUT.
     | PostCreateRedirect [Text] -- ^ Treat this request as a PUT, then redirect.
     | PostProcess (Handler s m ()) -- ^ Process as a POST, but don't redirect.
-    | PostProcessRedirect (Handler s m ByteString) -- ^ Process and redirect.
+    | PostProcessRedirect (Handler s m ()) (Handler s m ByteString) -- ^ Process and redirect.
```

Comment:
Perhaps incorrectly, I think I had previously assumed the user would do their "process" work as part of `Handler s m ByteString`. Thoughts on the tradeoffs between have the process effect intertwined with producing the bytestring for redirection?